### PR TITLE
set up server call enter, pickWinner, getPlayers for lottery demo

### PR DESCRIPTION
**Pull request by mikedoan**
## Issue
set up server call enter, pickWinner, getPlayers for lottery demo

### CODE REVIEW

File: node/node_genesis.go [hunk 1]
```diff
@@ -92,5 +94,6 @@ func AddNodeAddressesToGenesisAlloc(genesisAlloc core.GenesisAlloc) {
 		testBankFunds = testBankFunds.Mul(testBankFunds, big.NewInt(params.Ether))
 		address := common.HexToAddress(account.Address)
 		genesisAlloc[address] = core.GenesisAccount{Balance: testBankFunds}
+		utils.GetLogInstance().Info("****free money****", "address", address, "balance", testBankFunds)
```
Comment (mikedoan) [inline node/node_genesis.go]: trying to fund free money to these address for the demo. followings are just logs of the expectation we are looking at


INFO [03-15|22:28:04.574] ****free money****                       port=9000 ip=127.0.0.1 address=0x1a3e7a44ee21101d7D64FBf29B0F6F1fc295F723 balance=1000000000000000000000
INFO [03-15|22:28:04.575] ****free money****                       port=9000 ip=127.0.0.1 address=0x10A02A0a6e95a676AE23e2db04BEa3D1B8b7ca2E balance=1000000000000000000000
INFO [03-15|22:28:04.575] ****free money****                       port=9000 ip=127.0.0.1 address=0x3e881F6C36A3A14a2D1816b0A5471d1caBB16F33 balance=1000000000000000000000
INFO [03-15|22:28:04.575] ****free money****                       port=9000 ip=127.0.0.1 address=0x9d72989b68777a1f3FfD6F1DB079f1928373eE52 balance=1000000000000000000000
INFO [03-15|22:28:04.575] ****free money****                       port=9000 ip=127.0.0.1 address=0x67957240b6eB045E17B47dcE98102f09aaC03435 balance=1000000000000000000000
INFO [03-15|22:28:04.575] ****free money****                       port=9000 ip=127.0.0.1 address=0xf70fBDB1AD002baDF19024785b1a4bf6F841F558 balance=1000000000000000000000
INFO [03-15|22:28:04.575] ****free money****                       port=9000 ip=127.0.0.1 address=0x3f1A559be93C9456Ca75712535Fd522f5EC22c6B balance=1000000000000000000000
INFO [03-15|22:28:04.575] ****free money****                       port=9000 ip=127.0.0.1 address=0xedD257B4e0F5e7d632c737f4277e93b64DC268FC balance=1000000000000000000000
INFO [03-15|22:28:04.575] ****free money****                       port=9000 ip=127.0.0.1 address=0x66A74477FC1dd0F4924ed943C1d2F1Dece3Ab138 balance=1000000000000000000000
INFO [03-15|22:28:04.575] ****free money****                       port=9000 ip=127.0.0.1 address=0x04178CdbCe3a9Ff9Ea385777aFc4b78B3E745281 balance=1000000000000000000000

File: node/demo_contract.go [hunk 2]
```diff
@@ -43,41 +43,108 @@ func (node *Node) AddLotteryContract() {
...
+		}
+		address := crypto.PubkeyToAddress(key.PublicKey)
+		balances = append(balances, node.GetBalanceOfAddress(address))
+	}
+	return players, balances
+}
+
+// GetResult get current players and their balances.
+func (node *Node) GetResult() (players []string, balances []uint64) {
+	return node.GetResult2()
```
Comment (mikedoan) [inline node/demo_contract.go]: GetResult calls GetResult2

File: node/demo_contract.go [hunk 3]
```diff
@@ -43,41 +43,108 @@ func (node *Node) AddLotteryContract() {
...
-	} else {
-		utils.GetLogInstance().Error("Unable to call enter method", "error", err)
+		return nil
 	}
+	utils.GetLogInstance().Error("Unable to call enter method", "error", err)
+	return err
+}
+
+// GetResult2 get current players and their balances.
+func (node *Node) GetResult2() (players []string, balances []uint64) {
```
Comment (mikedoan) [inline node/demo_contract.go]: GetResult2 actually checks the balances of DemoAccounts which we tried to fund 1000 ether each

File: node/node_handler.go [hunk 4]
```diff
@@ -290,6 +290,16 @@ func (node *Node) PostConsensusProcessing(newBlock *types.Block) {
 	node.UpdateStakingList(node.QueryStakeInfo())
 	node.printStakingList()
 	node.blockchain.StoreNewShardState(newBlock, &node.CurrentStakes)
+
+	node.Debug()
+}
+
+// Debug --
+func (node *Node) Debug() {
+	players, balances := node.GetResult()
```
Comment (mikedoan) [inline node/node_handler.go]: Look at the GetResult above

File: node/node_handler.go [hunk 5]
```diff
@@ -290,6 +290,16 @@ func (node *Node) PostConsensusProcessing(newBlock *types.Block) {
...
+	for i := range players {
+		utils.GetLogInstance().Info("******RESULT****", "player", players[i], "balance", balances[i])
```
Comment (mikedoan) [inline node/node_handler.go]: INFO [03-15|22:30:53.355] ******RESULT****                         port=9000 ip=127.0.0.1 player=0x1a3e7a44ee21101d7D64FBf29B0F6F1fc295F723 balance=3875820019684212736
INFO [03-15|22:30:53.355] ******RESULT****                         port=9000 ip=127.0.0.1 player=0x10A02A0a6e95a676AE23e2db04BEa3D1B8b7ca2E balance=3875820019684212736
INFO [03-15|22:30:53.355] ******RESULT****                         port=9000 ip=127.0.0.1 player=0x3e881F6C36A3A14a2D1816b0A5471d1caBB16F33 balance=3875820019684212736
INFO [03-15|22:30:53.355] ******RESULT****                         port=9000 ip=127.0.0.1 player=0x9d72989b68777a1f3FfD6F1DB079f1928373eE52 balance=3875820019684212736
INFO [03-15|22:30:53.355] ******RESULT****                         port=9000 ip=127.0.0.1 player=0x67957240b6eB045E17B47dcE98102f09aaC03435 balance=3875820019684212736
INFO [03-15|22:30:53.356] ******RESULT****                         port=9000 ip=127.0.0.1 player=0xf70fBDB1AD002baDF19024785b1a4bf6F841F558 balance=3875820019684212736
INFO [03-15|22:30:53.356] ******RESULT****                         port=9000 ip=127.0.0.1 player=0x3f1A559be93C9456Ca75712535Fd522f5EC22c6B balance=3875820019684212736
INFO [03-15|22:30:53.356] ******RESULT****                         port=9000 ip=127.0.0.1 player=0xedD257B4e0F5e7d632c737f4277e93b64DC268FC balance=3875820019684212736
INFO [03-15|22:30:53.356] ******RESULT****                         port=9000 ip=127.0.0.1 player=0x66A74477FC1dd0F4924ed943C1d2F1Dece3Ab138 balance=3875820019684212736
INFO [03-15|22:30:53.356] ******RESULT****                         port=9000 ip=127.0.0.1 player=0x04178CdbCe3a9Ff9Ea385777aFc4b78B3E745281 balance=3875820019684212736
INFO [03-15|22:30:53.356] ******RESULT****                         port=9000 ip=127.0.0.1 player=0x46C61d50874A7A06D29FF89a710AbBD0856265be balance=3875820019684212736
INFO [03-15|22:30:53.356] ******RESULT****                         port=9000 ip=127.0.0.1 player=0xfE9BABE6904C28E31971337738FBCBAF8c72873e balance=3875820019684212736
INFO [03-15|22:30:53.356] ******RESULT****                         port=9000 ip=127.0.0.1 player=0x3f78622de8D8f87EAa0E8b28C2851e2450E91250 balance=3875820019684212736
INFO [03-15|22:30:53.356] ******RESULT****                         port=9000 ip=127.0.0.1 player=0xd2Cb501B40D3a9a013A38267a4d2A4Cf6bD2CAa8 balance=3875820019684212736

where each balance is not 1000 ether

File: api/proto/message/client.go [hunk 6]
```diff
@@ -37,12 +39,10 @@ func (client *Client) Close() {
 }
 
 // Process processes message.
-func (client *Client) Process(message *Message) *Response {
+func (client *Client) Process(message *Message) (*Response, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 	defer cancel()
+	utils.GetLogInstance().Info("***SENDING MESSAGE", "message", message)
```
Comment (LeoHChen) [inline api/proto/message/client.go]: can we reduce the log per message?

File: api/proto/message/server.go [hunk 7]
```diff
@@ -19,11 +23,59 @@ const (
...
+	GetResult                       func(string) ([]string, []*big.Int)
+	CreateTransactionForPickWinner  func() error
 }
 
 // Process processes the Message and returns Response
 func (s *Server) Process(ctx context.Context, message *Message) (*Response, error) {
+	if message.GetType() != MessageType_LOTTERY_REQUEST {
+		return &Response{}, ErrWrongMessage
+	}
+	utils.GetLogInstance().Info("Recieved:", "message", message)
```
Comment (LeoHChen) [inline api/proto/message/server.go]: too much debug message.

File: internal/utils/contract/constants.go [hunk 8]
```diff
@@ -85,20 +85,6 @@ var DemoAccounts = [...]DeployAccount{
 
 // NewNodeAccounts is the accounts used for new node to stake and join the network.
 var NewNodeAccounts = [...]DeployAccount{
-	{Address: "0x1a3e7a44ee21101d7D64FBf29B0F6F1fc295F723", Private: "27978f895b11d9c737e1ab1623fde722c04b4f9ccb4ab776bf15932cc72d7c66", Public: "0x1a3e7a44ee21101d7D64FBf29B0F6F1fc295F723"},
```
Comment (LeoHChen) [inline internal/utils/contract/constants.go]: why remove address here?

**Pull request closed without merging**